### PR TITLE
Add onBack lambda

### LINKED ISSUE #25: Override hardware back press
Sometimes it is useful to have a full control over the HW back press behavior when modal sheet is opened. For example when sheet displays two states (e.g. via AnimatedContent) and on back press it is expected to return from 2nd state to the 1st one. Now it is not possible, because `PopupLayout` always overrides `dispatchKeyEvent` (`onDismiss` lambda is always not-null), which means that `BackHandler`s added in sheet content or in the sheet scope are not invoked.

I would suggest to add an ability to override and / or disable default back press behavior.

### PR DESCRIPTION
**Pull request by oleksandrbalan**
Also add a new example how back button could be overridden.

Closes #25

https://user-images.githubusercontent.com/20944869/234996025-7e512ad4-da7b-48f8-bd70-af0568fb5391.mp4

### CODE REVIEW

File: modalsheet/src/main/java/eu/wewox/modalsheet/ModalSheet.kt [hunk 1]
```diff
@@ -45,6 +47,7 @@ public fun <T> ModalSheet(
     data: T?,
     onDataChange: (T?) -> Unit,
     cancelable: Boolean = true,
+    onBack: (() -> Unit)? = { onDataChange(null) },
```
Comment (wooodenleg) [inline modalsheet/src/main/java/eu/wewox/modalsheet/ModalSheet.kt:50]: nit: would name it something like `onSystemBack` or `onBackPressed`  to stay consistent with `Activity::onBackPressed`

**Review by wooodenleg — APPROVED** [review]

File: modalsheet/src/main/java/eu/wewox/modalsheet/ModalSheet.kt (hunk 1 above)
Comment (oleksandrbalan) [reply modalsheet/src/main/java/eu/wewox/modalsheet/ModalSheet.kt:50]: Actually I was inspired by `BackHandler` 😅 

https://developer.android.com/reference/kotlin/androidx/activity/compose/package-summary#BackHandler(kotlin.Boolean,kotlin.Function0)

Comment (wooodenleg) [reply modalsheet/src/main/java/eu/wewox/modalsheet/ModalSheet.kt:50]: Yeah, but in context of  `BackHandler` it makes more sense than when you call `ModalSheet`. But I don't insist on it so feel free to resolve

Comment (oleksandrbalan) [reply modalsheet/src/main/java/eu/wewox/modalsheet/ModalSheet.kt]: Oh, I see, makes sense 👍 

I will rename it to `onSystemBack`

**Pull request merged**